src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const clonedState = { ...state };
+  const stateArr = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(clonedState, action.extraData);
+        stateArr.push({ ...clonedState });
```

Comment:
Check point 4 in the checklist - [DON'T REPEAT YOURSELF] - If you perform the same action in all switch cases - do it just once afterward.
You can push the object once after the switch block.